### PR TITLE
Allow run in parallel, switch to remote diff

### PR DESCRIPTION
**Pull request by jojand**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -37,6 +38,7 @@ const ARGOCD_SERVER_URL = core.getInput('argocd-server-url');
 const ARGOCD_TOKEN = core.getInput('argocd-token');
 const VERSION = core.getInput('argocd-version');
 const EXTRA_CLI_ARGS = core.getInput('argocd-extra-cli-args');
+const INSECURE = core.getInput('insecure');
```
Comment (coufalja) [inline src/main.ts:41]: What about that parallelisation limit?

File: src/main.ts [hunk 2]
```diff
@@ -196,45 +204,55 @@ _Updated at ${new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angele
...
+        const command = `app diff ${app.metadata.name} --revision=${github.context.payload.pull_request?.head?.sha}`;
+        try {
+          core.info(`Running: argocd ${command}`);
+          // ArgoCD app diff will exit 1 if there is a diff, so always catch,
+          // and then consider it a success if there's a diff in stdout
+          // https://github.com/argoproj/argo-cd/issues/3588
+          await argocd(command);
+        } catch (e) {
+          const res = e as ExecResult;
+          core.info(`stdout: ${res.stdout}`);
```
Comment (coufalja) [inline src/main.ts:234]: Remove?

File: src/main.ts [hunk 3]
```diff
@@ -196,45 +204,55 @@ _Updated at ${new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angele
...
+        }
+      })
+    );
+  });
+
+  await Promise.all(input);
+
+  diffs = diffs.sort((a, b) => {
+    return a.app.metadata.name.localeCompare(b.app.metadata.name);
   });
```
Comment (coufalja) [inline src/main.ts]: Could be `diffs.sort((a, b) => a.app.metadata.name.localeCompare(b.app.metadata.name))`

**Review by coufalja — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 2 above)
Comment (jojand) [reply src/main.ts:234]: I'd keep it for debugging

**Review by coufalja — APPROVED** [review]

**Pull request merged**